.github/workflows/reusable-deploy-otc.yml

```diff
@@ -0,0 +1,88 @@
+name: Deploy
+
+on:
+  workflow_call:
+    inputs:
+      imageTag:
+        required: true
+        type: string
+      environment:
+        required: true
+        type: string
+    secrets:
+      DOCKER_REGISTRY:
+        required: true
+      REGISTRY_USER:
+        required: true
+      REGISTRY_PWD:
+        required: true
+      ERICA_HOST:
+        required: true
+      ERICA_SSH_KEY:
+        required: true
+      CERT_PIN:
+        required: true
+      QUEUE_URL:
+        required: true
+      ERICA_DATABASE_URL:
+        required: true
+      DEPLOY_KEY:
+        required: true
+      ARGOCD_SERVER:
+        required: true
+      ARGOCD_PIPELINE_PASSWORD:
+        required: true
+
+env:
+  NAMESPACE: erica-${{ inputs.environment }}
+
+jobs:
+  deploy:
+    name: Deploy
+
+    runs-on: ubuntu-latest
+
+    environment: ${{ inputs.environment }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Login to container repository
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ secrets.DOCKER_REGISTRY }}
+          username: ${{ secrets.REGISTRY_USER }}
+          password: ${{ secrets.REGISTRY_PWD }}
+
+      - name: Pull deploy image
+        run: docker pull ${{ secrets.DOCKER_REGISTRY }}/steuerlotse_deployment
+        shell: bash
+
+      - name: Deploy to Kubernetes via ArgoCD
+        uses: digitalservice4germany/github-actions/argocd-deploy@e5a88da4a7c71de837e1d813ba6d90faf5ca0c83
+        with:
+          environment: ${{ inputs.environment }}
+          version: ${{ inputs.imageTag }}
+          deploying_repo: erica
+          infra_repo: erica-infra
+          deploy_key: ${{ secrets.DEPLOY_KEY }}
+          app: erica-${{ inputs.environment }}
+          argocd_pipeline_password: ${{ secrets.ARGOCD_PIPELINE_PASSWORD }}
+          argocd_server: ${{ secrets.ARGOCD_SERVER }}
```

Comment:
Is this updating the image tag in the Erica-infra?

Reply:
It should, but I just noticed that it is not doing it in the way that we need it to… I'll have to replace this with some custom steps